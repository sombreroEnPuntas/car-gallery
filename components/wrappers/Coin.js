import styled from 'styled-components'
import { colors } from '.'

const Coin = styled.div`
  width: 21px;
  height: 12px;
  background: ${colors.coin.main};
  box-shadow:
    /* 
    * Inner Background Color
    */ 0 -6px 0 -2px
      ${colors.coin.main},
    0 6px 0 -2px ${colors.coin.main}, 0 -12px 0 -4px ${colors.coin.main},
    0 12px 0 -4px ${colors.coin.main}, 0 -18px 0 -8px ${colors.coin.main},
    0 18px 0 -8px ${colors.coin.main}, 0 -24px 0 -12px ${colors.coin.main},
    0 24px 0 -12px ${colors.coin.main},
    /* 
    * Inner Border
    */ -2px 0 0 0 ${colors.coin.border},
    2px 0 0 0 ${colors.coin.border}, 0 -4px 0 0 ${colors.coin.border},
    0 4px 0 0 ${colors.coin.border}, 0 -10px 0 -2px ${colors.coin.border},
    0 10px 0 -2px ${colors.coin.border}, 0 -14px 0 -4px ${colors.coin.border},
    0 14px 0 -4px ${colors.coin.border}, 0 -20px 0 -8px ${colors.coin.border},
    0 20px 0 -8px ${colors.coin.border}, 0 -26px 0 -12px ${colors.coin.border},
    0 26px 0 -12px ${colors.coin.border},
    /* 
    * Middle Border 
    */ -4px 0 0 0 ${colors.coin.accent},
    4px 0 0 0 ${colors.coin.accent}, 0 -2px 0 2px ${colors.coin.accent},
    0 2px 0 2px ${colors.coin.accent}, 0 -8px 0 0px ${colors.coin.accent},
    0 8px 0 0px ${colors.coin.accent}, 0 -12px 0 -2px ${colors.coin.accent},
    0 12px 0 -2px ${colors.coin.accent}, 0 -16px 0 -4px ${colors.coin.accent},
    0 16px 0 -4px ${colors.coin.accent}, 0 -22px 0 -8px ${colors.coin.accent},
    0 22px 0 -8px ${colors.coin.accent}, 0 -28px 0 -12px ${colors.coin.accent},
    0 28px 0 -12px ${colors.coin.accent},
    /*
    * Outer Border
    */ -6px 0 0 0 ${colors.coin.border},
    6px 0 0 0 ${colors.coin.border}, 0 0 0 4px ${colors.coin.border},
    0 -6px 0 2px ${colors.coin.border}, 0 6px 0 2px ${colors.coin.border},
    0 -10px 0 0 ${colors.coin.border}, 0 10px 0 0 ${colors.coin.border},
    0 -14px 0 -2px ${colors.coin.border}, 0 14px 0 -2px ${colors.coin.border},
    0 -18px 0 -4px ${colors.coin.border}, 0 18px 0 -4px ${colors.coin.border},
    0 -24px 0 -8px ${colors.coin.border}, 0 24px 0 -8px ${colors.coin.border},
    0 -30px 0 -12px ${colors.coin.border}, 0 30px 0 -12px ${colors.coin.border};
`

export default Coin
