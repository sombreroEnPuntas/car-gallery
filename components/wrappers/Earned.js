import styled from 'styled-components'
import { animations } from '.'

const Earned = styled.div`
  animation: ${animations.rotateIn} 1.4s ease-out forwards,
    ${animations.fadeOut} 2s ease-out forwards;

  position: absolute;
`

export default Earned
