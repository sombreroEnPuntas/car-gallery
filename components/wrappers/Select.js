import styled from 'styled-components'
import { breakpoints, colors, fonts } from '.'

const Select = styled.select`
  appearance: none;
  background-color: ${({ disabled }) => (disabled ? colors.disabled : 'white')};
  ::-ms-expand {
    display: none;
  }
  border: none;
  box-shadow: 0 4px ${({ status }) => colors[status || 'black'].shadow},
    0 -4px ${({ status }) => colors[status || 'black'].shadow},
    4px 0 ${({ status }) => colors[status || 'black'].shadow},
    -4px 0 ${({ status }) => colors[status || 'black'].shadow};
  font-family: ${fonts.fontFamily};
  line-height: calc(2 * ${fonts.lineHeight});
  margin: 4px;
  outline-color: ${({ status }) => colors[status || 'black'].outline};
  padding: 0.5rem 1rem;
  text-transform: capitalize;
  width: 100%;

  @media (min-width: ${breakpoints['mobile']}) {
    font-size: ${fonts.lineHeight};
    line-height: ${fonts.lineHeight};
  }
`

export default Select
