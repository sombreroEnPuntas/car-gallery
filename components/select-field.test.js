// @flow
import React from 'react'
import { shallow } from 'enzyme'

// Dependencies
import Coin from './wrappers/Coin'
import Earned from '../components/wrappers/Earned'
import Select from '../components/wrappers/Select'

// Tested unit
import TestedComponent from './select-field'

const handleUpdate = jest.fn()
const getProps = customProps => ({
  disabled: false,
  handleUpdate,
  name: 'dinosaur-name',
  valid: false,
  value: '',
  values: ['t-rex', 'brontosaurus'],
  ...customProps,
})

describe('DropdownField', () => {
  describe.each`
    status          | Component | event                         | value
    ${'char typed'} | ${Select} | ${{ target: { value: 'r' } }} | ${'r'}
    ${'char typed'} | ${Select} | ${{ target: { value: 'x' } }} | ${'x'}
  `('$status', ({ Component, event, value }) => {
    const wrapper = shallow(<TestedComponent {...getProps({ value })} />)

    it(`handles the event properly`, () => {
      wrapper
        .find(Component)
        .simulate(Component === Select ? 'change' : 'click', event)

      expect(handleUpdate).toHaveBeenCalledWith(value, getProps().name)
    })
  })

  it(`matches snapshot`, () => {
    const wrapper = shallow(<TestedComponent {...getProps()} />)

    expect(wrapper).toMatchSnapshot()
  })

  it(`shows valid status`, () => {
    const wrapper = shallow(<TestedComponent {...getProps({ valid: true })} />)

    expect(
      wrapper
        .find(Select)
        .dive()
        .prop('status')
    ).toBe('success')

    expect(wrapper.find(Coin)).toHaveLength(1)
    expect(wrapper.find(Earned)).toHaveLength(1)
  })
})
