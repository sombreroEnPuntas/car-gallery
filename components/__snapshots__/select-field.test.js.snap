// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownField matches snapshot 1`] = `
<FormLineWrap>
  <Select
    autoComplete="off"
    disabled={false}
    handleUpdate={
      [MockFunction] {
        "calls": Array [
          Array [
            "r",
            "dinosaur-name",
          ],
          Array [
            "x",
            "dinosaur-name",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    name="dinosaur-name"
    onChange={[Function]}
    status={null}
    type="text"
    value=""
  >
    <SelectItem
      id="dinosaur-name-placeholder"
      key="dinosaur-name-placeholder"
    />
    <SelectItem
      id="dinosaur-name-undefined"
      key="dinosaur-name-undefined"
    />
  </Select>
</FormLineWrap>
`;
